src/Application/Actions/Donations/Update.php

```diff
@@ -215,6 +215,9 @@ private function addData(Donation $donation, HttpModels\Donation $donationData,
         );
 
         if ($donation->getPsp() === 'stripe') {
+            $transactionId = $donation->getTransactionId();
+            \assert(is_string($transactionId));
```

Comment:
It's not immediately obvious to me how we know at this point that the donation has a transaction ID, but its also not clear if there's a reason for this class to change in this PR.

Since it looks like this pattern is repeated a few times it might be worth adding a method to Donation that returns the transaction ID as a string and throws if it's null, with docs and a name to make it clear it should only be called when we know there's a transaction ID.

Reply:
If we're going to start using `assert` outside of test code it would be good to know how it's configured in production. For options see https://www.php.net/manual/en/ini.core.php#ini.zend.assertions and https://www.php.net/manual/en/info.configuration.php#ini.assert.exception

Or we could use a library like https://github.com/beberlei/assert or instead https://github.com/webmozarts/assert . These both come with annotations that Psalm understands on the assertion methods, and will always throw if the assertion is false.

Reply:
@bdsl The reason for changing this class is that the continued use of `StripeClientInterface` is IMO a trap which makes writing future psalm-failing code and then having to fix it again more likely.

Reply:
Right. We're not the first to have this issue - see https://github.com/stripe/stripe-php/issues/970 from three years ago. Seems like `StripeClientInterface` may be of dubious value. I think if we just change the type from StripeClientInterface to StripeClient and that makes the psalm error appear it's fine to add it to the baseline and leave the code `$this->stripeClient->paymentIntents->confirm($donation->getTransactionId());` as it was, since it's not really a new error it's just an old one in a different form.

Reply:
@bdsl I was mainly following your lead in introducing `\assert` to fix errors – it's already in Production or en route via `UpdatePaymentMethod` & co.

I'm 99% sure our current config changes no defaults, so asserts will be evaluated and give warnings but not terminate. This seems fine to me as a starting point: it means that introducing stricter code won't break things unexpectedly but that we will have a clear way to monitor the AWS dashboards and search logs for unexpected assert failures.

Reply:
> Right. We're not the first to have this issue - see [stripe/stripe-php#970](https://github.com/stripe/stripe-php/issues/970) from three years ago. Seems like `StripeClientInterface` may be of dubious value. I think if we just change the type from StripeClientInterface to StripeClient and that makes the psalm error appear it's fine to add it to the baseline and leave the code `$this->stripeClient->paymentIntents->confirm($donation->getTransactionId());` as it was, since it's not really a new error it's just an old one in a different form.

Think we overlapped commenting here. As the pre-existing class is now green with the stricter approach, I suggest we keep the changes.

Reply:
Used a getter instead